mono/mini/interp/interp.c

```diff
@@ -750,7 +750,12 @@ static InterpMethodArguments* build_args_from_sig (MonoMethodSignature *sig, Mon
 			break;
 #if SIZEOF_VOID_P == 4
 		case MONO_TYPE_I8:
-			g_assert (0);
+#ifdef TARGET_ARM
+			/* pairs begin at even registers */
+			if (margs->ilen & 1)
+				margs->ilen++;
+#endif
```

Comment:
it's already behind `TARGET_ARM`.  are there differences between armv5, armv7 etc.?